src/OpenLoco/Map/Tile.h

```diff
@@ -206,12 +204,13 @@ namespace OpenLoco::Map
     private:
         uint8_t _4;
         uint8_t _5;
-        uint8_t _6;
-        uint8_t _7;
+        uint16_t _6;
 
     public:
         bool has_40() const { return (_type & 0x40) != 0; }
         bool hasStationElement() const { return (_type & 0x80) != 0; }
+        uint8_t colour() const { return _6 >> 11; }
+        void setColour(Colour_t colour) { _6 = (_6 & 0x7FF) | (colour << 11); }
```

Comment:
This does not have the desired effect. Compare:
```
loc_42F0E7:
movzx   ax, bl     ; bl = colour
shl     ax, 11
and     word ptr [esi+6], 7FFh
or      [esi+6], ax
```
If I run the original routine and inspect the building element in the tile inspector, I'm seeing the last (7th) byte be updated instead. The second to last byte appears to stay 0.

Our byte index numbering starts at 0, so I expected a uint16_t at pos _6 to cover both bytes. However, the colour isn't updating at all. What am I missing?

Reply:
Is `(colour << 11)` evaluating correctly? Not sure if `colour` needs to be casted to a larger integer first or not.

Reply:
Thanks for the suggestion. The issue is in `updateHeadquartersColourAtTile`; I was using `auto` instead of `auto&`. Will push the fix shortly.